drivers/cc1200/cc1200-rxtx.c

```diff
@@ -0,0 +1,430 @@
+/*
+ * Copyright (C) 2017 TU Berlin DAI Labor
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @ingroup     drivers_cc1200
+ * @{
+ * @file
+ * @brief       Functions for packet reception and transmission on cc1200 devices
+ *
+ * @author      Anon Mall <mail@anonmall.com>
+ * @}
+ */
+
+#include <errno.h>
+#include <stdio.h>
+#include <string.h>
+
+#include "cc1200.h"
+#include "cc1200-spi.h"
+#include "cc1200-internal.h"
+#include "cc1200-interface.h"
+#include "cc1200-defines.h"
+
+#include "periph/gpio.h"
+#include "irq.h"
+
+#include "kernel_types.h"
+#include "msg.h"
+#include "xtimer.h"
+
+#include "cpu_conf.h"
+#include "cpu.h"
+
+#include "log.h"
+#include "led.h"
+
+#define ENABLE_DEBUG (0)
+#include "debug.h"
+
+static void _rx_abort(cc1200_t *dev)
+{
+    gpio_irq_disable(dev->params.int_pin);
+
+    cc1200_strobe(dev, CC1200_SIDLE);       /* Switch to IDLE (should already be)... */
+    cc1200_strobe(dev, CC1200_SFRX);        /* ...for flushing the RX FIFO */
+
+    cc1200_switch_to_rx(dev);
+}
+
+
+
+static void _rx_read_data(cc1200_t *dev, void (*callback)(void *), void *arg)
+{
+    DEBUG("%s:%s:%u\n", RIOT_FILE_RELATIVE, __func__, __LINE__);
+
+    uint8_t overflow = cc1200_strobe(dev, CC1200_SNOP);
+    if ((overflow & 0x70) == 0x60) {
+        DEBUG("%s:%s:%u rx overflow\n", RIOT_FILE_RELATIVE, __func__, __LINE__);
+        _rx_abort(dev);
+        return;
+    }
+
+
+    uint8_t fifo = cc1200_read_reg(dev, CC1200_NUM_RXBYTES);
+
+    if (!fifo) {
+        DEBUG("%s:%s:%u rx fifo is empty\n", RIOT_FILE_RELATIVE, __func__, __LINE__);
+        gpio_irq_enable(dev->params.int_pin);
+        return;
+    }
+
+    cc1200_pkt_buf_t *pkt_buf = &dev->pkt_buf;
+    if (!pkt_buf->pos) {
+        pkt_buf->pos = 1;
+
+        pkt_buf->packet.length = cc1200_read_reg(dev, CC1200_RXFIFO);
+
+        /* Possible packet received, RX -> IDLE (0.1 us) */
+        dev->cc1200_statistic.packets_in++;
```

Comment:
Isn't this the location where the `MODULE_NETSTATS_L2` should be fed?

Reply:
You're right. Good point.

Reply:
This is done in the `_recv(...)` function in `cc1200-netdev.c`, as there is no access to the `ieee802154_netdev_t` device at that point.